File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        
        if nums[l] < nums[r]:
            return nums[l]

        while l <= r:
            m = (l + r) // 2
            if nums[m] > nums[r]:
                if nums[m] < nums[m + 1]:
                    l = m + 1
                else:
                    return nums[m + 1]
            elif nums[m] < nums[l]:
                if nums[m] > nums[m - 1]:
                    r = m - 1
                else:
                    return nums[m]
            elif l == r:
                return nums[m]
            

File: Data Structures & Algorithms/frequency-of-the-most-frequent-element/submission-1.rs
impl Solution {
    pub fn max_frequency(mut nums: Vec<i32>, k: i32) -> i32 {
        nums.sort();
        let (mut l, mut total, mut length) = (0, 0, 0);
        for r in 0..nums.len() {
            total += nums[r];

            if ((r - l + 1) as i32) * nums[r] > total + k {
                total -= nums[l];
                l += 1;
            }
            length = max((r - l + 1) as i32, length);
        }
        length
    }
}
